test/Microsoft.AspNetCore.Mvc.Core.Test/ModelBinding/Binders/ComplexTypeModelBinderTest.cs

```diff
@@ -350,6 +350,52 @@ public void CreateModel_InstantiatesInstanceOfMetadataType()
         }
 
         [Fact]
+        public void CreateModel_ForStructModelType_AsTopLevelObject_ThrowsException()
+        {
+            // Arrange
+            var bindingContext = new DefaultModelBindingContext
+            {
+                ModelMetadata = GetMetadataForType(typeof(PointStruct)),
+                IsTopLevelObject = true
+            };
+            var binder = CreateBinder(bindingContext.ModelMetadata);
+
+            // Act & Assert
+            var exception = Assert.Throws<InvalidOperationException>(() => binder.CreateModelPublic(bindingContext));
+            Assert.Equal(
+                string.Format(
+                    "Could not create an instance of type '{0}'. Model bound complex types must not be abstract or " +
+                    "value types and must have a parameterless constructor.",
+                    typeof(PointStruct).FullName),
+                exception.Message);
+        }
```

Comment:
Yes, this is a new error to make it easy diagnosable. In 1.0.0 following would have been the errors (which are not very helpful in case of value types I feel):

For value model type:
```
typeof(System.ArgumentException): Expression of type 'Microsoft.AspNetCore.Mvc.ModelBinding.Binders.ComplexTypeModelBinderTest+PointStruct' cannot be used for return type 'System.Object'
      Stack Trace:
           at System.Linq.Expressions.Expression.ValidateLambdaArgs(Type delegateType, Expression& body, ReadOnlyCollection`1 parameters, String paramName)
           at System.Linq.Expressions.Expression.Lambda[TDelegate](Expression body, String name, Boolean tailCall, IEnumerable`1 parameters)
           at System.Linq.Expressions.Expression.Lambda[TDelegate](Expression body, Boolean tailCall, IEnumerable`1 parameters)
           at System.Linq.Expressions.Expression.Lambda[TDelegate](Expression body, ParameterExpression[] parameters)
        C:\github\mvc\src\Microsoft.AspNetCore.Mvc.Core\ModelBinding\Binders\ComplexTypeModelBinder.cs(346,0): at Microsoft.AspNetCore.Mvc.ModelBinding.Binders.ComplexTypeModelBinder.CreateModel(ModelBindingContext bindingContext)
        C:\github\mvc\test\Microsoft.AspNetCore.Mvc.Core.Test\ModelBinding\Binders\ComplexTypeModelBinderTest.cs(1409,0): at Microsoft.AspNetCore.Mvc.ModelBinding.Binders.ComplexTypeModelBinderTest.TestableComplexTypeModelBinder.CreateModelPublic(ModelBindingContext bindingContext)
        C:\github\mvc\test\Microsoft.AspNetCore.Mvc.Core.Test\ModelBinding\Binders\ComplexTypeModelBinderTest.cs(364,0): at Microsoft.AspNetCore.Mvc.ModelBinding.Binders.ComplexTypeModelBinderTest.<>c__DisplayClass15_0.<CreateModel_ForStructModelType_AsTopLevelObject_ThrowsException>b__0()
```
----------------
For reference model type:
```
typeof(System.ArgumentException): Type 'Microsoft.AspNetCore.Mvc.ModelBinding.Binders.ComplexTypeModelBinderTest+PointStruct' does not have a default constructor
      Parameter name: type
      Stack Trace:
           at System.Linq.Expressions.Expression.New(Type type)
        C:\github\mvc\src\Microsoft.AspNetCore.Mvc.Core\ModelBinding\Binders\ComplexTypeModelBinder.cs(346,0): at Microsoft.AspNetCore.Mvc.ModelBinding.Binders.ComplexTypeModelBinder.CreateModel(ModelBindingContext bindingContext)
        C:\github\mvc\test\Microsoft.AspNetCore.Mvc.Core.Test\ModelBinding\Binders\ComplexTypeModelBinderTest.cs(1409,0): at Microsoft.AspNetCore.Mvc.ModelBinding.Binders.ComplexTypeModelBinderTest.TestableComplexTypeModelBinder.CreateModelPublic(ModelBindingContext bindingContext)
        C:\github\mvc\test\Microsoft.AspNetCore.Mvc.Core.Test\ModelBinding\Binders\ComplexTypeModelBinderTest.cs(364,0): at Microsoft.AspNetCore.Mvc.ModelBinding.Binders.ComplexTypeModelBinderTest.<>c__DisplayClass15_0.<CreateModel_ForStructModelType_AsTopLevelObject_ThrowsException>b__0()```

> but it has to be mentioned.

You mean we need a comment here?